heaps/min_heap.py

```diff
@@ -57,18 +69,44 @@ def heap_up(self, index):
             property is reestablished.
             
             This could be **very** helpful for the add method.
-            Time complexity: ?
-            Space complexity: ?
+            Time complexity: O(log n)
+            Space complexity: O(1)
+
+            check my value vs parent's value
+            if parent is larger 
+                swap with parent
+                recursively call heap_up from new location
         """
-        pass
+        if index == 0:
+            return
+        
+        parent = (index - 1) // 2
+        store = self.store
+        if store[parent].key > store[index].key:
+            self.swap(parent, index)
+            self.heap_up(parent)
 
     def heap_down(self, index):
         """ This helper method takes an index and 
             moves the corresponding element down the heap if it's 
             larger than either of its children and continues until
             the heap property is reestablished.
         """
-        pass
+        left_child = index * 2 + 1
```

Comment:
👀 This is identical to the instructor solution and has no additions that would indicate to me that time was spent time understanding this approach.